apps/web/components/eventtype/LocationsAutocomplete.tsx

```diff
@@ -0,0 +1,108 @@
+import Image from "next/image";
+import React, { useCallback, useEffect, useRef, useState } from "react";
+
+import { useDebounce } from "@calcom/lib/hooks/useDebounce";
+import { Input } from "@calcom/ui";
+
+import { useOnClickOutside } from "@lib/hooks/useOnClickOutside";
+
+interface LocationsAutocompleteProps {
+  value: string | undefined;
+  placeholder: string;
+  name: string;
+  required?: boolean;
+
+  onSave: (place: string) => void;
+}
+
+function LocationsAutocomplete({
```

Comment:
we also use EditLocationDialog while editing location from bookings page so we can use this component there also. (can be done later)

Reply:
Right, I'll take a deeper look into the codebase to see the different use cases. And make the component more generic to reuse it accordingly